src/control-plane/backend/lambda/api/common/stack-params-valid.ts

```diff
@@ -389,34 +392,43 @@ async function _checkForRedshiftServerless(redshiftType: string, pipeline: IPipe
 }
 
 async function _checkVpcEndpointsForIsolatedSubnets(pipeline: IPipeline, vpcId: string,
-  privateSubnets: ClickStreamSubnet[], allSubnets: ClickStreamSubnet[]) {
-  const isolatedSubnets = privateSubnets.filter(subnet => subnet.type == SubnetType.ISOLATED);
-  if (isolatedSubnets.length > 0) {
-    const isolatedSubnetsAZ = getSubnetsAZ(isolatedSubnets);
-    const vpcEndpoints = await describeVpcEndpoints(pipeline.region, vpcId);
-    const vpcEndpointSecurityGroups: string[] = _getEndpointSecurityGroups(vpcEndpoints);
-
-    const vpcEndpointSecurityGroupRules = await describeSecurityGroupsWithRules(pipeline.region, vpcEndpointSecurityGroups);
-
-    for (let privateSubnet of privateSubnets) {
-      if (privateSubnet.type === SubnetType.ISOLATED) {
-        validateVpcEndpoint(
-          pipeline.region,
-          allSubnets,
-          isolatedSubnetsAZ,
-          privateSubnet,
-          vpcEndpoints,
-          vpcEndpointSecurityGroupRules,
-          [
-            's3',
-            'logs',
-          ]);
-        _validateEndpointsForModules(pipeline, allSubnets, isolatedSubnetsAZ, privateSubnet, vpcEndpoints, vpcEndpointSecurityGroupRules);
-      }
+  selectedPrivateSubnets: ClickStreamSubnet[], allSubnets: ClickStreamSubnet[]) {
+  const vpcEndpoints = await describeVpcEndpoints(pipeline.region, vpcId);
+  const vpcEndpointSecurityGroups: string[] = _getEndpointSecurityGroups(vpcEndpoints);
```

Comment:
Should we check the security group per endpoint?

Reply:
Fetch all SG once filter by endpoint in function:validateVpcEndpoint.